.github/workflows/build-and-test.yml

```diff
@@ -117,7 +117,6 @@ jobs:
       - name: Download charm
         uses: actions/download-artifact@v4
         with:
-          name: charm-artifact-${{ matrix.part }}
```

Comment:
question: Is this removal done for dloading all charm artifacts rather than just for `matrix-part` ?

Reply:
The name was removed because we're using the pattern, and thus using the name doesn't make sense.

Reply:
ack